src/react-hooks-nesting-walker/is-hook-call.ts

```diff
@@ -20,11 +21,14 @@ export function isHookCall(
   ruleOptions: RuleOptions,
 ) {
   if (isIdentifier(expression) && isHookIdentifier(expression)) {
-    return true;
+    return !ruleOptions[ignoredFunctionsOptionName].has(expression.text);
   } else if (
     isPropertyAccessExpression(expression) &&
     isHookIdentifier(expression.name)
   ) {
+    if (ruleOptions[ignoredFunctionsOptionName].has(expression.name.text)) {
```

Comment:
I am not sure about allowing rule users to ignore hooks from the React namespace. It seems to me that adding the `ignored-functions` option is for functions that are not hooks themselves, but their name looks like one.

When it comes to real hooks, I reckon this tslint rule should not allow ignoring their possibly incorrect usage globally. In such exceptional cases, a `tslint:disable` could be used

I would rather not allow ignoring hooks from the `React` namespace 🙂 Let me know what you think